boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,355 @@
+ * @param efuse_spi_handle Handle to the Efuse's SPI peripheral
+ * @param chip_select_port Handle to Efuse's chip-select GPIO port
+ * @param chip_select_pin Efuse's chip-select GPIO pin
+ * @return EXIT_CODE_OK if the read was successful
+ *         EXIT_CODE_TIMEOUT if the SPI read timed-out
+ */
+static ExitCode Io_Efuse_ReadFromEfuse(
+    uint16_t *         tx_data,
+    uint16_t *         rx_data,
+    SPI_HandleTypeDef *efuse_spi_handle,
+    GPIO_TypeDef *     chip_select_port,
+    uint16_t           chip_select_pin);
+
+struct Efuse_Context *Io_Efuse_Create(
+    SPI_HandleTypeDef *const spi_handle,
+    GPIO_TypeDef *           nss_port,
+    uint16_t                 nss_pin,
+    GPIO_TypeDef *           fsob_port,
+    uint16_t                 fsob_pin,
+    GPIO_TypeDef *           fsb_port,
+    uint16_t                 fsb_pin,
+    GPIO_TypeDef *           channel0_port,
+    uint16_t                 channel0_pin,
+    GPIO_TypeDef *           channel1_port,
+    uint16_t                 channel1_pin)
+{
+    assert(spi_handle != NULL);
+
+    struct Efuse_Context *efuse_context = malloc(sizeof(struct Efuse_Context));
+    assert(efuse_context != NULL);
+
+    efuse_context->hspi            = spi_handle;
+    efuse_context->nss_port        = nss_port;
+    efuse_context->nss_pin         = nss_pin;
+    efuse_context->fsob_port       = fsob_port;
+    efuse_context->fsob_pin        = fsob_pin;
+    efuse_context->fsb_port        = fsb_port;
+    efuse_context->fsb_pin         = fsb_pin;
+    efuse_context->channel0_port   = channel0_port;
+    efuse_context->channel0_pin    = channel0_pin;
+    efuse_context->channel1_port   = channel1_port;
+    efuse_context->channel1_pin    = channel1_pin;
+    efuse_context->wdin_bit_to_set = true;
+
+    return efuse_context;
+}
+
+void Io_Efuse_Destroy(struct Efuse_Context *e_fuse)
+{
+    free(e_fuse);
+}
+
+void Io_Efuse_EnableChannel0(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel0_port, e_fuse->channel0_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel0(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel0_port, e_fuse->channel0_pin, GPIO_PIN_RESET);
+}
+
+void Io_Efuse_EnableChannel1(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel1_port, e_fuse->channel1_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel1(const struct Efuse_Context *const e_fuse)
+{
+    HAL_GPIO_WritePin(
+        e_fuse->channel1_port, e_fuse->channel1_pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     selection,
+    struct Efuse_Context *const e_fuse)
+{
+    uint16_t register_value = 0x0000;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_ReadRegister(SI_GCR_ADDR, &register_value, e_fuse));
+
+    CLEAR_BIT(register_value, (CSNS1_EN_MASK | CSNS0_EN_MASK));
```

Comment:
I think a small comment for what we're doing with the value read here would be nice

Reply:
This line reads as :
CLEAR_BIT on the CSNS1 and CSN0 EN MASK 
Does a comment add anything useful? Similar to the below comment

Reply:
Oh i just meant a small comment about why we need to manipulate the bit this way lol

Reply:
Oh i see, added a comment to make it more clear